source/common/http/http3/conn_pool.cc

```diff
@@ -119,8 +119,7 @@ Http3ConnPoolImpl::createClientConnection(Quic::QuicStatNames& quic_stat_names,
                                           OptRef<Http::HttpServerPropertiesCache> rtt_cache,
                                           Stats::Scope& scope) {
   std::shared_ptr<quic::QuicCryptoClientConfig> crypto_config =
-      dynamic_cast<Quic::QuicClientTransportSocketFactory&>(host_->transportSocketFactory())
-          .getCryptoConfig();
+      host_->transportSocketFactory().getCryptoConfig();
```

Comment:
For my own knowledge, the proxy transport socket factory returns null here, then how would the conn pool act upon this return value?

Reply:
isn't it a reference, not a pointer?